fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/db2/Db2ResourceReferenceDAO.java

```diff
@@ -149,7 +149,13 @@ protected void doCommonTokenValuesUpsert(String paramList, Collection<CommonToke
                 ps.setInt(a++, tv.getCodeSystemId());
             }
 
-            ps.executeUpdate();
+            try {
+                ps.executeUpdate();
+            } catch (SQLException x) {
+                logger.throwing("Db2ResourceReferenceDAO", "doCommonTokenValuesUpsert", x);
+                logger.fine("Retrying the execution in case of synchronization issues");
+                ps.executeUpdate();
+            }
```

Comment:
Retry logic

Reply:
I opted not to use common_token_values ON CONFLICT IGNORE as there is a small possibility of ignoring a relevant ID, and I wasn't sure if that was OK or not.